Build the applications provided by the container when  the image is created

penguineer opened the pull request:
Instead of relying on external sources build the applications when the container is created.

This fixes #7

penguineer committed:
Compile software when image is built

penguineer committed:
Remove detect-apt-proxy

The apt proxy concept does not work well with CI environments.

penguineer commented:
@christf Bitte mal testen, ob das so läuft.

christf commented:
it builds and looks ok locally.  I like the multi-staged builds and the simplification by removing apt proxy
Maybe this is now the time when we re-purpose our 2nd batman gateway to a babel gateway so we have again 2 to work with.

penguineer commented:
> it builds and looks ok locally.

Did you run it? I saw that the wg-broker script did not find wg, but it might be due to my setup?

> I like the multi-staged builds and the simplification by removing apt proxy 

Thanks :)

> Maybe this is now the time when we re-purpose our 2nd batman gateway to a babel gateway so we have again 2 to work with.

We need to change Babel upstream first, this uses the 2nd batman gateway.

christf commented:
just a question: how do you use your apt cache in that setup?

penguineer commented:
> just a question: how do you use your apt cache in that setup?

I don't. The complexity introduced into the CI system is not worth the gain, especially since we are not layering the image build.

I would rather consider doing some availability/version check in the run script to avoid installing software when it is not needed.

24367dfa commented on README.md:
```diff
@@ -19,12 +19,8 @@ package feed. This allows source-specific routes being transported properly.
 The build can make use of an apt cache.
```
is this still accurate?

penguineer replied:
It is not, thank you for pointing that out.

I have also remove the reference to a specific babeld version.

penguineer committed:
Remove apt-cache reference from README

penguineer committed:
Remove reference to babeld release

We are no longer building babeld 1.9 specifically, but everything is checked
out from master for now.

24367dfa reviewed (approved):
lgtm
